jest/transform.js

```diff
@@ -18,7 +18,8 @@ const maybeTsConfig = tryParseTsConfig();
 module.exports.transform = Object.fromEntries(
   Object.entries(defaults.transform).map(([key, value]) => [
     key,
-    value === TS_JEST_NAME
+    (typeof value === 'string' && value === TS_JEST_NAME) ||
+    (Array.isArray(value) && value[0] === TS_JEST_NAME)
```

Comment:
Should we do something like this for the array scenario:

```typescript
[
  TS_JEST_PATH,
  { ...value[1], isolatedModules: value[1].isolatedModules ?? maybeTsConfig.options.isolatedModules ?? true }
]
```

Reply:
Good call, I've split up the different types since it was getting messy